okio/src/test/kotlin/okio/ByteStringTest.kt

```diff
@@ -16,21 +16,67 @@
 
 package okio
 
+import okio.ByteString.Companion.decodeBase64
 import okio.ByteString.Companion.decodeHex
+import okio.ByteString.Companion.encodeUtf8
+import okio.internal.commonAsUtf8ToByteArray
 import kotlin.test.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertSame
 import kotlin.test.assertTrue
+import kotlin.test.fail
+
+class CommonByteStringTest {
+  class Platform : AbstractByteStringTest(ByteStringFactory.PLATFORM)
+  class Okio : AbstractByteStringTest(ByteStringFactory.OKIO)
+}
+
+interface ByteStringFactory {
+  fun decodeHex(hex: String): ByteString
+  fun encodeUtf8(s: String): ByteString
+
+  companion object {
+    val PLATFORM: ByteStringFactory = object : ByteStringFactory {
+      override fun decodeHex(hex: String) = hex.decodeHex()
+      override fun encodeUtf8(s: String) = s.encodeUtf8()
+    }
+    val OKIO: ByteStringFactory = object : ByteStringFactory {
```

Comment:
worth a small bit of doc to explain the consequences of two implementations?

Reply:
Yes, I'll document both and explain what each is trying to test.